app/src/main/java/com/android/voyageur/ui/trip/assistant/Assistant.kt

```diff
@@ -0,0 +1,123 @@
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Button
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.android.voyageur.model.activity.Activity
+import com.android.voyageur.model.activity.extractActivitiesFromJson
+import com.android.voyageur.model.trip.TripsViewModel
+import com.android.voyageur.model.trip.UiState
+import com.android.voyageur.ui.navigation.NavigationActions
+import com.android.voyageur.ui.trip.activities.ActivityItem
+import com.android.voyageur.ui.trip.activities.ButtonType
+import com.android.voyageur.ui.trip.schedule.TopBarWithImageAndText
+
+@Composable
+fun AssistantScreen(tripsViewModel: TripsViewModel, navigationActions: NavigationActions) {
+  var result by rememberSaveable { mutableStateOf("placeholderResult") }
+  val uiState by tripsViewModel.uiState.collectAsState()
+  var prompt by rememberSaveable { mutableStateOf("") }
+  var activities by remember { mutableStateOf(emptyList<Activity>()) }
+
+  val trip =
+      tripsViewModel.selectedTrip.value ?: return Text(text = "No trip selected. Should not happen")
+
+  Scaffold(
+      modifier = Modifier.testTag("assistantScreen"),
+      content = { pd ->
+        Column {
+          TopBarWithImageAndText(trip, navigationActions, "Ask the AI assistant!", trip.name)
+          Row(modifier = Modifier.padding(all = 16.dp)) {
+            TextField(
+                value = prompt,
+                label = { Text("prompt") },
+                onValueChange = { prompt = it },
+                modifier =
+                    Modifier.testTag("AIRequestTextField")
+                        .weight(0.8f)
+                        .padding(end = 16.dp)
+                        .align(Alignment.CenterVertically))
+            Button(
+                onClick = { tripsViewModel.sendActivitiesPrompt(trip, prompt) },
+                modifier = Modifier.testTag("AIRequestButton").align(Alignment.CenterVertically)) {
+                  Text(text = "go")
```

Comment:
Minor suggestion: Consider disabling the button while the results are loading to prevent multiple unnecessary API calls.

Reply:
Additionally, consider disabling the button when the prompt is blank. This would provide better input validation on the UI level and prevent unnecessary API calls with invalid or empty prompts.

Reply:
I disabled the button when the ui state is loading, thanks for the suggestion! However I think that users should be able to generate results even without a prompt: it would just generate activities for the trip with that title, without other requirements. In the future when we'll add other features to this screen and clearly define functionalities I will make sure to display how an empty prompt behaves, but for the moment I think that it's okay to leave it like that.

Reply:
You make a valid point, and I hadn’t considered that perspective. Allowing users to generate results without a prompt does seem reasonable, especially if it defaults to generating activities for the trip with that title, without other requirements